ts/Dashboard/EditMode/Fullscreen.ts

```diff
@@ -0,0 +1,148 @@
+import Dashboard from '../Dashboard.js';
+import U from '../../Core/Utilities.js';
+
+const { addEvent } = U;
+
+class Fullscreen {
+    /* *
+     *
+     *  Static Functions
+     *
+     * */
+
+    /**
+     * Prepares the dashboard class to support fullscreen.
+     * @param {typeof_Dashboard} DashboardClass
+     *        The dashboard class to decorate with fullscreen support.
+     */
+    public static compose(DashboardClass: Dashboard): void {
+        // Initialize fullscreen
+        DashboardClass.fullscreen = new Fullscreen(DashboardClass);
+    }
+
+    /* *
+    *
+    *  Static Properties
+    *
+    * */
+
+
+    /* *
+    *
+    *  Constructor
+    *
+    * */
+    constructor(DashboardClass: Dashboard) {
+        this.isOpen = false;
+        this.dashboard = DashboardClass;
+    }
+
+    /* *
+    *
+    *  Properties
+    *
+    * */
+
+    public dashboard: Dashboard;
+    public isOpen: boolean;
+    public unbindFullscreenEvent?: Function;
+
+
+    /* *
+    *
+    *  Functions
+    *
+    * */
+
+
+    /**
+     * Toggles displaying the dashboard in fullscreen mode.
+     */
+    public toggle(): void {
+        const fullscreen = this,
+            isOpen = this.isOpen;
+
+        fullscreen[isOpen ? 'close' : 'open']();
+    }
+
+    /**
+     * Display dashboard in fullscreen.
+     */
+    public open(): void {
+        const fullscreen = this,
+            dashboard = fullscreen.dashboard;
+
+        // Handle exitFullscreen() method when user clicks 'Escape' button.
+        const unbindChange = addEvent(
+            dashboard.container.ownerDocument, // dashboard's document
+            'fullscreenchange',
+            function (): void {
+                if (fullscreen.isOpen) {
+                    fullscreen.isOpen = false;
+                    fullscreen.close();
+                } else {
+                    fullscreen.isOpen = true;
```

Comment:
You can set the flag into `fullscreen.setButtonText()`

Reply:
I feel like `setButtonText` should do no more and no less than what the name states.